libbeat/beat/config.go

```diff
@@ -0,0 +1,43 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package beat
+
+import (
+	"github.com/elastic/beats/libbeat/keystore"
+	ucfg "github.com/elastic/go-ucfg"
+)
+
+// ConfigOpts returns ucfg config options with a resolver linked to the current keystore.
+// TODO: Refactor to allow insert into the config option array without having to redefine everything
+func ConfigOpts(store keystore.Keystore) []ucfg.Option {
```

Comment:
Let's not have these in `libbeat/beat` package. This package is supposed to be mostly stable providing some minimal common API for beats developers (No bc breaking changes if possible). Items with "TODO" + "Refactoring" should not land here.

Also, let's not leak the ucfg package. Libbeat is supposed to hide/wrap ucfg package.

Reply:
What do you propose to do to make sure we actually correctly initialize the resolvers?

Reply:
Let me take a look at that we pass to libbeat.